design/FY2022/NFP-Python-HVAC-Diagram.md

```diff
@@ -0,0 +1,85 @@
+# Python HVAC Diagram #
+
+Jason DeGraw, Piljae Im, and Niraj Kunwar
+
+Oak Ridge National Laboratory
+
+Original Date: November 18, 2021
+
+## Justification for New Feature ##
+
+The HVAC-Diagram program is a Fortran utility that can be used to generate
+an SVG representation of the HVAC system in a model based on the BND file
+generated by EnergyPlus. The program has limited flexibility and the SVG
+output is not correctly rendered by all SVG rendering libraries. The SVG format
+[1] is a large and complex format with many capabilities but is somewhat
+overpowered for this application.
+
+## Overview ##
+
+A Python replacement for the utility will be developed to produce output similar
+to the functionality of the original utility. While SVG output will be considered,
+prioirity will be placed on production of output that conveys the same information
+in a more reliable manner.
+
+## Approach ##
+
+The BND output from EnergyPlus will be used to generate several forms of output:
+
+* One or more DOT language [2] files. This format is used by the
+  open source Graphviz package to display and manipulate graphs. Other formats
+  will be considered depending on development team feedback.
```

Comment:
I like the DOT approach but I am concerned that the resulting diagram will be too jumbled. Is their some way with the DOT format to specify the structure of the diagrams such that airloops are shown in one area and water loops in another with links between them?

BTW, I just used a similar approach of a Python script creating a DOT file here: 

https://github.com/JasonGlazer/jsonSchemaToDot

and you can see that the result (in the PDF) can be a little jumbled.

Reply:
There some ways to set attributes of the graph, but rendering is usually going to be up to the program that does the rendering and I would expect that most renderings would be jumbled. Before the log4j thing happened, I usually used the Gephi program to look at graphs, and depending on the layout algorithm the output can be pretty messy. In any case, the DOT output isn't going to be the default and will definitely be more of an "expert" thing. I did it originally to make sure I was getting the connections out of the BND correctly and decided to leave it in.